src/ClientData/CaptureDataTest.cpp

```diff
@@ -123,4 +132,47 @@ TEST_F(CaptureDataTest, UpdateScopeStatsIsCorrect) {
                    GetStats(kDurationsForSecondId, kSecondVariance));
 }
 
+TEST_F(CaptureDataTest, VarianceIsCorrectForLongDurations) {
+  for (TimerInfo timer : kTimerInfos) {
+    timer.set_end(timer.end() + kLargeInteger);
+    capture_data_.UpdateScopeStats(timer);
+  }
+
+  capture_data_.UpdateScopeStats(kTimerInfoWithInvalidScopeId);
+
+  ORBIT_LOG("VARIANCE %f", capture_data_.GetScopeStatsOrDefault(kFirstId).variance_ns());
+  EXPECT_NEAR(capture_data_.GetScopeStatsOrDefault(kFirstId).variance_ns(), kFirstVariance, 1.0);
+  EXPECT_NEAR(capture_data_.GetScopeStatsOrDefault(kSecondId).variance_ns(), kSecondVariance, 1.0);
+}
+
+TEST_F(CaptureDataTest, VarianceIsCorrectOnScimitarDataset) {
+  std::filesystem::path path = orbit_test::GetTestdataDir() / "scimitar_variance_and_durations.csv";
+  const ErrorMessageOr<std::string> file_content_or_error = orbit_base::ReadFileToString(path);
+  EXPECT_TRUE(file_content_or_error.has_value());
+  const std::string& file_content = file_content_or_error.value();
+  const std::vector<absl::string_view> tokens = absl::StrSplit(file_content, '\n');
+
+  double expected_variance;
+  EXPECT_TRUE(absl::SimpleAtod(*tokens.begin(), &expected_variance));
+
+  std::vector<TimerInfo> timers;
+  std::transform(std::begin(tokens) + 1, std::end(tokens) - 1, std::back_inserter(timers),
+                 [](const absl::string_view& line) {
+                   uint64_t duration = std::stoull(std::string(std::begin(line), std::end(line)));
+                   TimerInfo timer;
+                   timer.set_function_id(kFirstId);
+                   timer.set_start(0);
+                   timer.set_end(duration);
+                   return timer;
+                 });
+
+  for (const TimerInfo& timer : timers) {
+    capture_data_.UpdateScopeStats(timer);
+  }
+
+  const double actual_variance = capture_data_.GetScopeStatsOrDefault(kFirstId).variance_ns();
+
+  EXPECT_LE(abs(actual_variance / expected_variance - 1.0), 1e-5);
```

Comment:
Can you use EXPECT_NEAR here?

Reply:
`EXPECT_NEAR` checks if the two values are close as absolute values. Here relative difference makes more sense.
`EXPECT_NEAR(actual_variance, expected_variance, 1'000'000'000)` would be strange.

Reply:
Okay. Thanks.